staking/programs/integrity-pool/src/state/pool.rs

```diff
@@ -221,35 +236,56 @@ impl PoolData {
             }
             self.del_state[i] = next_del_state;
             self.self_del_state[i] = next_self_del_state;
+
+            // for every event that was missed, create a reward event using del_state after update
+            // which corresponds to the del_state of all the epochs after last_updated_epoch
+            self.create_reward_events_for_publisher(
+                self.last_updated_epoch + 1,
+                current_epoch,
+                i,
+                publisher_caps.caps[cap_index].cap,
+            )?;
+
+            i += 1;
         }
 
-        // for every event that was missed, create a reward event using del_state after update
-        // which corresponds to the del_state of all the epochs after last_updated_epoch
-        for epoch in self.last_updated_epoch + 1..current_epoch {
-            self.create_reward_event(publisher_caps, y, epoch)?;
+        for j in 0..(publisher_caps.num_publishers as usize) {
+            if existing_publishers[j / 8] & (1 << (j % 8)) == 0 {
+                require_gt!(MAX_PUBLISHERS, i, IntegrityPoolError::TooManyPublishers);
```

Comment:
perhaps it's better to handle this by just not adding a publisher if the array is full? Just to avoid this error blocking everything

Reply:
Done